views/js/chatControl.js

```diff
@@ -2,8 +2,44 @@ var chatActiveRoom = "general";
 var chatRefreshTimeMs = 30000;
 
 var refreshChat = function () {
-  document.getElementById("chat_messages").innerHTML = fetchChatHTML();
-  setChatScrollToBottom();
+  fetch('./chat/' + chatActiveRoom)
+    .then(function (response) {
+      return response.json();
+    })
+    .then(function (chatList) {
+      var chatHTML = "";
+      chatList.forEach(function (element) {
+        chatHTML += `<div class="
+    individual_chat_message"><p><span class="
+    author">${element.name}</span>${element.message}</p></div>`;
+      });
+      document.getElementById("chat_messages").innerHTML = chatHTML;
+      setChatScrollToBottom();
+    });
+}
+
+var postMessageToChat = function () {
+  var name = document.getElementById("name_input").value;
+  var message = document.getElementById("message_input").value;
+  document.getElementById("message_input").value = "";
+  fetch('/chat/' + chatActiveRoom, {
+      method: 'POST',
+      headers: {
+        'Content-Type': 'application/json',
+      },
+      body: JSON.stringify({
+        "name": name,
+        "message": message
+      })
+    })
+    .then((response) => response.text())
+    .then((responseText) => {
```

Comment:
Since you don't use `responseText`, why not simply call `setTimeout` in the first closure?

Reply:
Done https://github.com/knative-portability/Kubercade/pull/22/commits/a245f8e5651072a53fca0919a807d492cb10b02b